lightblue/src/main/java/org/esbtools/eventhandler/lightblue/locking/LockingRoutePolicy.java

```diff
@@ -0,0 +1,66 @@
+package org.esbtools.eventhandler.lightblue.locking;
+
+import java.io.IOException;
+
+import javax.annotation.Nullable;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Route;
+import org.apache.camel.support.RoutePolicySupport;
+
+public class LockingRoutePolicy extends RoutePolicySupport {
+
+    private final String identifier;
+
+    private final LockStrategy lockStrategy;
+
+    private @Nullable LockedResource<String> lock;
+
+    public LockingRoutePolicy(String identifier, LockStrategy lockStrategy) {
+        this.identifier = identifier;
```

Comment:
Nitpicky, but it would be more consistent with lock strategy interface if `identifier` was named `resourceId`, perhaps I even had this wrong originally.